File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {

        double magnitude = getMagnitude(xOfDart, yOfDart);

        if (magnitude > 10.0)
            return 0;

        if (magnitude > 5.0)
            return 1;

        if (magnitude > 1.0)
            return 5;

        return 10;
    }

    private double getMagnitude(double xOfDart, double yOfDart) {
        return Math.sqrt((xOfDart * xOfDart) + (yOfDart * yOfDart));
    }
}

File: solutions/go/darts/1/darts.go
package darts

import "math"

func Score(x, y float64) int {
	dart_pos := math.Sqrt(x*x + y*y)

	switch {
	case dart_pos <= 1.0:
		return 10
	case dart_pos <= 5.0:
		return 5
	case dart_pos <= 10.0:
		return 1
	default:
		return 0
	}
}
